processor/src/main/java/org/mapstruct/ap/internal/model/assignment/AdderWrapper.java

```diff
@@ -61,7 +61,7 @@ public AdderWrapper( Assignment decoratedAssignment, List<Type> thrownTypesToExc
     public Set<Type> getImportTypes() {
         Set<Type> imported = new HashSet<Type>();
         imported.addAll( super.getImportTypes() );
-        imported.add( getSourceType() );
+        imported.add( getSourceType().getTypeParameters().get( 0 ) );
```

Comment:
@filiphr an adder calls the add method on the target type. The argument is an element type from a collection. The collection is supposed to be set by a setter.

```java

      @Override
    public void toExistingTarget(Source source, Target target) {
        if ( source == null ) {
            return;
        }

        try {
            if ( source.getPets() != null ) {
                for ( String list : source.getPets() ) {
                    target.addPet( petMapper.toPet( list ) );
                }
            }
        }
        catch ( CatException e ) {
            throw new RuntimeException( e );
        }
        catch ( DogException e ) {
            throw new RuntimeException( e );
        }
    }
```

But.. the variable name should follow the element name as well.. I'll fix that as well.